### DISCUSSION TITLE
IP-ranges

### DISCUSSION THREAD

**Discussion by trs91km** [netbox-community/netbox#12846] [General]
I am trying to update IPV6 IP ranges, getting error message as **"Defined range exceeds maximum supported size (4294967295)"**, is there a limitation? now how I need to update this?

**Comment by candlerb** [comment]
Please can you specify:
* Which Netbox version you're using
* What actual values you're trying to enter in the range (replace first 32 bits with `2001:db8:` for privacy)

But looking in v3.5.3 source, it does look like a range is limited as per that message:

```
            # Validate maximum size
            MAX_SIZE = 2 ** 32 - 1
            if int(self.end_address.ip - self.start_address.ip) + 1 > MAX_SIZE:
                raise ValidationError(f"Defined range exceeds maximum supported size ({MAX_SIZE})")
```

This implies you should be able to do `2001:db8::1/128` to `2001:db8::ffff:ffff/128`

**Comment by dylanjamesdev** [comment]
I'd like to be able too add /40 blocks....

**Reply by candlerb** [reply]
For a /40 block, you should create a Prefix of type "Container" (since this is larger than a single /64 prefix, and other prefixes will be allocated within it).

The purpose of an IP Range is to allocate a range of addresses *within* a prefix for a particular purpose (e.g. a DHCP pool) which doesn't have to lie on a prefix boundary.

**Reply by AlmeidaCaio** [reply]
I tried to do the creation for /64 and it gave me the same error:
```
{   "__all__": [     "Defined range exceeds maximum supported size (4294967295)"   ] }
```
I'm using NetBox Community v4.3.6-Docker-3.3.0; therefore, should I use prefix too?
